norse/torch/functional/izhikevich.py

```diff
@@ -0,0 +1,173 @@
+import torch
+from typing import NamedTuple, Tuple
+
+from norse.torch.functional.threshold import threshold
+
+
+class IzhikevichParameters(NamedTuple):
+    """Parametrization of av Izhikevich neuron
+    Parameters:
+        a (float): time scale of the recovery variable u. Smaller values result in slower recovery in 1/ms
+        b (float): sensitivity of the recovery variable u to the subthreshold fluctuations of the membrane potential v. Greater values couple v and u more strongly resulting in possible subthreshold oscillations and low-threshold spiking dynamics
+        c (float): after-spike reset value of the membrane potential in mV
+        d (float): after-spike reset of the recovery variable u caused by slow high-threshold Na+ and K+ conductances in mV
+        sq (float): constant of the v squared variable in mV/ms
+        mn (float): constant of the v variable in 1/ms
+        bias (float): bias constant in mV/ms
+        v_th (torch.Tensor): threshold potential in mV
+        tau_inv (float) : inverse time constant in 1/ms
+        method (str): method to determine the spike threshold
+                      (relevant for surrogate gradients)
+        alpha (float): hyper parameter to use in surrogate gradient computation
+    """
+
+    a: float
+    b: float
+    c: float
+    d: float
+    sq: float = 0.04
+    mn: float = 5
+    bias: float = 140
+    v_th: float = 30
+    tau_inv: float = 250
+    method: str = "super"
+    alpha: float = 100.0
+
+
+class IzhikevichState(NamedTuple):
+    """State of a Izhikevich neuron
+    Parameters:
+        v (torch.Tensor): membrane potential
+        u (torch.Tensor): membrane recovery variable
+    """
+
+    v: torch.Tensor
+    u: torch.Tensor
+
+
+class IzhikevichSpikingBehaviour(NamedTuple):
+    """Spiking behaviour of a Izhikevich neuron
+    Parameters:
+        p (IzhikevichParameters) : parameters of the Izhikevich neuron model
+        s (IzhikevichState) : state of the Izhikevich neuron model
+    """
+
+    p: IzhikevichParameters
+    s: IzhikevichState
+
+
+p_ts = IzhikevichParameters(a=0.02, b=0.2, c=-65, d=6)
```

Comment:
I can see why you'd want to have the parameters stored like this, but this means that the `p_ts` variable gets exposed in the `izhikevich.py` module upon import. If you'd like to keep it this way, I'd suggest you rename it to something more meaningful. Like `tonic_spiking_p`. That's a bit easier to read than `ts`.

Reply:
Same comment applies to the cases below.